### PR TITLE
feat: support assetSubtreeIds and assetExternalIds filters [DEVX-199]

### PR DESCRIPTION
**Pull request by polomani**
For:
- Events
- TimeSeries
- Assets (only assetSubtreeIds)

https://cognitedata.atlassian.net/browse/DEVX-199

### CODE REVIEW

File: src/types/types.ts [hunk 1]
```diff
@@ -295,8 +295,20 @@ export type AssetExternalId = ExternalId;
 export interface AssetFilter extends Limit {
   filter?: {
     name?: AssetName;
+    /**
+     * Return only the direct descendants of the specified assets.
+     */
     parentIds?: CogniteInternalId[];
+    /**
+     * Return only the direct descendants of the specified assets.
```
Comment (Meemaw) [inline src/types/types.ts:303]:
```suggestion
     * Returns only the direct descendants of the specified assets.
```

File: src/types/types.ts [hunk 2]
```diff
@@ -295,8 +295,20 @@ export type AssetExternalId = ExternalId;
 export interface AssetFilter extends Limit {
   filter?: {
     name?: AssetName;
+    /**
+     * Return only the direct descendants of the specified assets.
```
Comment (Meemaw) [inline src/types/types.ts:299]:
```suggestion
     * Returns only the direct descendants of the specified assets.
```

File: src/types/types.ts [hunk 3]
```diff
@@ -824,10 +853,17 @@ export interface FileFilter extends Limit {
      * Only include files that reference these specific asset IDs.
      */
     assetIds?: CogniteInternalId[];
+    /**
+     * Only include files that have a related asset in a tree rooted at any of these root assetIds.
```
Comment (Meemaw) [inline src/types/types.ts:860]: This comment is not consistent across types? In one place I see:

"The IDs of the root assets that the related assets should be children of." -- if the behaviour is the same they should probably have consistent comments.

**Review by Meemaw — APPROVED** [review]
There aren't many code changes so I don't have any real comments on that, but I added a few nitpicks about type comments -- try to be consistent with wording across resources.

File: src/utils.ts [hunk 4]
```diff
@@ -109,6 +109,8 @@ export function transformDateInResponse<T>(data: T): T {
     'uploadedTime',
     'deletedTime',
     'timestamp',
+    'sourceCreatedTime',
+    'sourceModifiedTime',
```
Comment (valgussev) [inline src/utils.ts:113]: How do these two related to the topic?

**Review by valgussev — APPROVED** [review]

File: src/utils.ts (hunk 4 above)
Comment (polomani) [reply src/utils.ts:113]: they are not really related, but they will go in a separate commit `feat(files): source created/modified time, rootAssetIds, assetSubtreeIds`, hence a separate item in the changelog as well.

Just wanted to save some time

Comment (valgussev) [reply src/utils.ts:113]: I see, than ready to merge

File: src/types/types.ts (hunk 3 above)
Comment (polomani) [reply src/types/types.ts:860]: agree

**Pull request merged**